### PR TITLE
#API-1123 - Update logger

### PR DESCRIPTION
**Pull request by k-macmillan**
# Description

Updated logging to allow for json in the lowers, specifically dev was broken due to a debug flag.
Tweaked LOG_FORMAT to apply one way for the api and another for celery (so we can see the worker).
Refactored the application name to be in-line with the the deployed service.
Modified filter to allow for `recordId` so logs can be tracked easier.

issue #[1123](https://github.com/department-of-veterans-affairs/notification-api/issues/1123)

## How Has This Been Tested?

[Deployed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/12146966137). Passes regression and logging looks good.

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/main/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

**Review by MackHalliday — APPROVED** [review]
Approved - Excited to see the updated log format.

File: notifications_utils/logging.py [hunk 1]
```diff
@@ -134,31 +142,37 @@ def get_handler(app):
...
-
     def filter(self, record):
-        record.requestId = self.request_id
+        # The else is for celery
+        if not getattr(record, 'requestId', ''):
+            record.requestId = RequestIdFilter._get_api_id() if has_request_context() else 'no-request-id'
         return record
+
+    @classmethod
+    def _get_api_id(self) -> str:
```
Comment (cris-oddball) [inline notifications_utils/logging.py:173]: nit: usually the first arg of a classmethod is cls, not self. just something to consider.

**Review by cris-oddball — APPROVED** [review]

**Review by MackHalliday — APPROVED** [review]
Reviewed new commits. Reapproved.

File: notifications_utils/logging.py [hunk 2]
```diff
@@ -134,31 +142,37 @@ def get_handler(app):
...
+        # The else is for celery
+        if not getattr(record, 'requestId', ''):
+            record.requestId = RequestIdFilter._get_api_id() if has_request_context() else 'no-request-id'
         return record
+
+    @staticmethod
+    def _get_api_id() -> str:
+        """Generate a request_id.
+
+        g is a global for this request. It is attached to the Flask and is only persisted for that request.
```
Comment (kalbfled) [inline notifications_utils/logging.py:176]:
```suggestion
        g is a Flask global for this request. It is attached to the Flask instance and is only persisted for that request.
```

Comment (k-macmillan) [reply notifications_utils/logging.py]: Changed.

**Pull request merged**